src/Npgsql/NpgsqlException.cs

```diff
@@ -45,6 +45,15 @@ public NpgsqlException(string message, Exception innerException)
         public NpgsqlException(string message)
             : base(message) { }
 
+        public virtual bool IsTransient
+        {
+            get
+            {
+                return this.InnerException != null && (this.InnerException.GetType() == typeof(System.IO.IOException));
```

Comment:
Some comments:
* Please turn this into an expression-bodied property
* I think it makes sense to also check for SocketException
* Why not simply use the `is` operator rather than calling `GetType()`?
* Also use the null coalescing operator (`??`) rather than checking for null